jerry-core/ecma/base/ecma-helpers-number.c

```diff
@@ -267,9 +267,25 @@ ecma_number_is_nan (ecma_number_t num) /**< ecma-number */
 ecma_number_t
 ecma_number_make_nan (void)
 {
-  return ecma_number_pack (false,
-                           (1u << ECMA_NUMBER_BIASED_EXP_WIDTH) - 1u,
-                           1u);
+#if ENABLED (JERRY_NUMBER_TYPE_FLOAT64)
+  union
+  {
+    double dbl;
+    uint64_t ui;
+  } f;
```

Comment:
Can we reuse this union for making +-Infinity value? If we could I'd move this into ecma-globals.h.

Reply:
Good idea. I did it. And removed all duplication of this union.